### PR TITLE
refactor(get_analyses_before)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Refactoring of the get analyses before function

### Added

- `get_analyses_for_case_and_pipeline_started_at_before`
- `get_analyses_for_pipeline_started_at_before`
-`get_analyses_started_at_before`
- Tests for these functions

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do cg clean hk-bundle-files

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: tests/store/api/status/test_store_api_status_analysis.py [hunk 1]
```diff
@@ -326,3 +326,92 @@ def test_one_of_one_sequenced_samples(
...
     assert test_case in cases
+
+
+def test_get_analyses_for_case_and_pipeline_before(
+    store_with_analyses_for_cases_not_uploaded_fluffy: Store,
+    timestamp_now: datetime,
+    pipeline: Pipeline = Pipeline.FLUFFY,
+    case_id: str = "yellowhog",
+):
+    """Test to get all analyses before a given date"""
```
Comment (diitaz93) [inline tests/store/api/status/test_store_api_status_analysis.py:337]:
```suggestion
    """Test to get all analyses before a given date."""
```

File: tests/store/api/status/test_store_api_status_analysis.py [hunk 2]
```diff
@@ -326,3 +326,92 @@ def test_one_of_one_sequenced_samples(
...
+        assert analysis.family.internal_id == case_id
+        assert analysis.pipeline == pipeline
+
+
+def test_get_analyses_for_case_before(
+    store_with_analyses_for_cases_not_uploaded_fluffy: Store,
+    timestamp_now: datetime,
+    case_id: str = "yellowhog",
+):
+    """Test to get all analyses before a given date"""
```
Comment (diitaz93) [inline tests/store/api/status/test_store_api_status_analysis.py:360]:
```suggestion
    """Test to get all analyses before a given date."""
```

File: tests/store/api/status/test_store_api_status_analysis.py [hunk 3]
```diff
@@ -326,3 +326,92 @@ def test_one_of_one_sequenced_samples(
...
+        assert analysis.started_at < timestamp_now
+        assert analysis.family.internal_id == case_id
+
+
+def test_get_analyses_for_pipeline_before(
+    store_with_analyses_for_cases_not_uploaded_fluffy: Store,
+    timestamp_now: datetime,
+    pipeline: Pipeline = Pipeline.FLUFFY,
+):
+    """Test to get all analyses for a pipeline before a given date"""
```
Comment (diitaz93) [inline tests/store/api/status/test_store_api_status_analysis.py:383]:
```suggestion
    """Test to get all analyses for a pipeline before a given date."""
```

File: tests/store/api/status/test_store_api_status_analysis.py [hunk 4]
```diff
@@ -326,3 +326,92 @@ def test_one_of_one_sequenced_samples(
...
+    for analysis in analyses:
+        assert analysis.started_at < timestamp_now
+        assert analysis.pipeline == pipeline
+
+
+def test_get_analyses_before(
+    store_with_analyses_for_cases_not_uploaded_fluffy: Store,
+    timestamp_now: datetime,
+):
+    """Test to get all analyses for a pipeline before a given date"""
```
Comment (diitaz93) [inline tests/store/api/status/test_store_api_status_analysis.py:404]:
```suggestion
    """Test to get all analyses for a pipeline before a given date."""
```

**Review by diitaz93 — APPROVED** [review]
👍

**Comment by ChrOertlin** [comment]
`cg clean hk-bundle-files -c acetetra -p mip-dna -o 500 -t fastq -d`
Cleaning analysis acetetra | 2017-11-23
Fetch version 2017-11-23 16:50:03 from bundle acetetra
Version found for bundle:acetetra; pipeline: mip-dna; date 2017-11-23 16:50:03
Fetching files from bundle acetetra
Fetching files with tags in [fastq]
Fetching files from version 2433
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/query.py:2577: SAWarning: Pathed join target File.version has already been joined to; skipping
  "been joined to; skipping" % prop
Process freed 0.0GB. Dry run: True

`cg clean hk-bundle-files -c acetetra -o 500 -t fastq -d`
Cleaning analysis acetetra | 2017-11-23
Fetch version 2017-11-23 16:50:03 from bundle acetetra
Version found for bundle:acetetra; pipeline: mip-dna; date 2017-11-23 16:50:03
Fetching files from bundle acetetra
Fetching files with tags in [fastq]
Fetching files from version 2433
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/query.py:2577: SAWarning: Pathed join target File.version has already been joined to; skipping
  "been joined to; skipping" % prop
Process freed 0.0GB. Dry run: True

`cg clean hk-bundle-files -p mip-dna -o 1000 -t fastq -d`
Cleaning analysis idealslug | 2017-09-26
Fetch version 2017-09-26 10:13:34 from bundle idealslug
Version found for bundle:idealslug; pipeline: mip-dna; date 2017-09-26 10:13:34
Fetching files from bundle idealslug
Fetching files with tags in [fastq]
Fetching files from version 1404
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/query.py:2577: SAWarning: Pathed join target File.version has already been joined to; skipping
  "been joined to; skipping" % prop
Cleaning analysis fondguppy | 2017-09-26
Fetch version 2017-09-26 10:10:36 from bundle fondguppy
Version found for bundle:fondguppy; pipeline: mip-dna; date 2017-09-26 10:10:36
Fetching files from bundle fondguppy
Fetching files with tags in [fastq]
Fetching files from version 1408
Cleaning analysis poeticworm | 2017-09-26
Fetch version 2017-09-26 10:05:21 from bundle poeticworm
Version found for bundle:poeticworm; pipeline: mip-dna; date 2017-09-26 10:05:21
Fetching files from bundle poeticworm
Fetching files with tags in [fastq]
Fetching files from version 1406
Cleaning analysis awaretiger | 2017-09-26
Fetch version 2017-09-26 09:58:27 from bundle awaretiger
Version found for bundle:awaretiger; pipeline: mip-dna; date 2017-09-26 09:58:27
Fetching files from bundle awaretiger
Fetching files with tags in [fastq]
Fetching files from version 1405
Cleaning analysis smoothbug | 2017-09-26
Fetch version 2017-09-26 09:48:00 from bundle smoothbug
Version found for bundle:smoothbug; pipeline: mip-dna; date 2017-09-26 09:48:00
Fetching files from bundle smoothbug
Fetching files with tags in [fastq]
Fetching files from version 1407
Cleaning analysis divinekoi | 2017-09-25
Fetch version 2017-09-25 09:34:19 from bundle divinekoi
Version found for bundle:divinekoi; pipeline: mip-dna; date 2017-09-25 09:34:19
Fetching files from bundle divinekoi
Fetching files with tags in [fastq]

**Pull request merged**